### DISCUSSION TITLE
Not use flutter

### DISCUSSION THREAD

**Discussion by iMonZ** [rustdesk/rustdesk#1056] [Polls]
Is it possible to not transition to flutter?
My experience is pretty bad. It sucks too many resources and also doesn’t look very modern or fitting as a desktop app! On android it’s perfect but on everything else, it’s the wrong framework. What do others think? Do someone agree or not?
I could help make a native iOS app and macOS OR for example using the MAUI framework which looks native on every platform and is easy to maintain as a piece of cake.

**Comment by Orholt** [comment]
Imagine them rewriting the whole codebase to switch to another language / framework.

It's just too much work. Flutter isn't that bad (I am not a flutter developer, I work with C#), it has its own pros and cons but at the moment I wouldn't expect it to change.

**Reply by iMonZ** [reply]
Avalonia, MAUI are good frameworks. Flutter is really bottlenecking your pc and has an awful touch to the design

**Reply by Orholt** [reply]
MAUI is kind of unstable, but I have few side projects on avalonia and it works very well.

**Comment by xavetar** [comment]
> 



> Imagine them rewriting the whole codebase to switch to another language / framework.
> 
> It's just too much work. Flutter isn't that bad (I am not a flutter developer, I work with C#), it has its own pros and cons but at the moment I wouldn't expect it to change.

Yes, rewrite the entire codebase. It needs to be rewritten from time to time to understand the bottlenecks of the system.

There are so many FFIs included in this project that it doesn’t smell like pure Rust, there’s still a lot of code written through the ass in the dependencies of the dependencies.

So many dependencies, it's pure horror. There are always a lot of problems with interfaces, why? It is necessary to write UI for a completely native stack for each OC and not use intermediary dependencies at all.

This project uses mostly cross-compilation capabilities - it's not viable without it. And this is not so much the fault of the developer as of the entire IT industry, instead of creating interaction interfaces, everyone breeds entities.

So much code has been written in the world that even books can envy, but the meaning is only? Most of the code (90%) is duplicates of other code.

**Comment by xavetar** [comment]
The problem of large amounts of code is huge cognitive complexity and a high entry threshold. Even in this project, several user interface options are used - this is a problem, instead of putting effort into one dependency, the developer sprays and writes "spells" for other operating systems, because there is a completely different way of working with the graphic subsystem and user interfaces (Fuck the Apple Documentation).

Take the same Dart and Flutter - well, shit, a lot of problems with them, one version completely breaks compatibility with another version - all this is not convenient. The very way of interacting with the code is not convenient when there are 2000 thousand lines - it's okay, when there are 25 of them - this is already a little different, the time to maneuver - takes a little more time, when there are more than 100 thousand of them - it will take a month, without taking into account the fact that some new functionality needs to be implemented. As a result, this results in millions of lines of code - which are somewhere - but whether they work correctly or work at all - no one knows.

We need to change the very approach to development, and I'm not talking about fucking CI/CD and containers or development teams, but to make it so that even one person can manipulate hundreds of thousands of lines of code with one click - adding new functionality or correcting logical errors.

Operating systems such as Unix/Linux only have 200-300 basic system calls that control the entire system.

The rest of the code - only interacts with the resulting data and change the order of data management, or change the idea of interacting with them.

**Comment by Unknown78** [comment]
The issue is still persists, random crash, random closing connections.

The name is RustDesk, but when will we have PureRust-Desk?

**Reply by xlionjuan** [reply]
You should create new discussion and describe your hardware, system environment and the way to reproduce.

**Reply by Unknown78** [reply]
I have in the past, https://github.com/rustdesk/rustdesk/issues/1729
I need to recheck it now though..

**Comment by congzhangzh** [comment]
try avaloniaui https://avaloniaui.net?

**Reply by Unknown78** [reply]
runtime again.. smh..
